tiledb/api/c_api/query_aggregate/test/unit_capi_query_aggregate.cc

```diff
@@ -0,0 +1,361 @@
+  uint64_t d2_size = 10 * sizeof(int64_t);
+
+  REQUIRE(
+      tiledb_query_set_data_buffer(ctx, query, "a", a, &a_size) == TILEDB_OK);
+
+  REQUIRE(
+      tiledb_query_set_data_buffer(ctx, query, "b", b, &b_size) == TILEDB_OK);
+
+  REQUIRE(
+      tiledb_query_set_data_buffer(ctx, query, "d1", d1, &d1_size) ==
+      TILEDB_OK);
+
+  REQUIRE(
+      tiledb_query_set_data_buffer(ctx, query, "d2", d2, &d2_size) ==
+      TILEDB_OK);
+
+  CHECK(tiledb_query_submit(ctx, query) == TILEDB_OK);
+
+  // Clean up
+  CHECK(tiledb_array_close(ctx, array) == TILEDB_OK);
+  tiledb_array_free(&array);
+  tiledb_query_free(&query);
+}
+
+void QueryAggregateFx::create_sparse_array(const std::string& array_name) {
+  // Create dimensions
+  uint64_t tile_extents[] = {2, 2};
+  uint64_t dim_domain[] = {1, 10, 1, 10};
+
+  tiledb_dimension_t* d1;
+  int rc = tiledb_dimension_alloc(
+      ctx, "d1", TILEDB_UINT64, &dim_domain[0], &tile_extents[0], &d1);
+  CHECK(rc == TILEDB_OK);
+  tiledb_dimension_t* d2;
+  rc = tiledb_dimension_alloc(
+      ctx, "d2", TILEDB_UINT64, &dim_domain[2], &tile_extents[1], &d2);
+  CHECK(rc == TILEDB_OK);
+
+  // Create domain
+  tiledb_domain_t* domain;
+  rc = tiledb_domain_alloc(ctx, &domain);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_domain_add_dimension(ctx, domain, d1);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_domain_add_dimension(ctx, domain, d2);
+  CHECK(rc == TILEDB_OK);
+
+  // Create attributes
+  tiledb_attribute_t* a;
+  rc = tiledb_attribute_alloc(ctx, "a", TILEDB_INT32, &a);
+  CHECK(rc == TILEDB_OK);
+  tiledb_attribute_t* b;
+  rc = tiledb_attribute_alloc(ctx, "b", TILEDB_INT32, &b);
+  CHECK(rc == TILEDB_OK);
+
+  // Create array schema
+  tiledb_array_schema_t* array_schema;
+  rc = tiledb_array_schema_alloc(ctx, TILEDB_SPARSE, &array_schema);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_cell_order(ctx, array_schema, TILEDB_ROW_MAJOR);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_tile_order(ctx, array_schema, TILEDB_ROW_MAJOR);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_capacity(ctx, array_schema, 4);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_domain(ctx, array_schema, domain);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_array_schema_add_attribute(ctx, array_schema, a);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_array_schema_add_attribute(ctx, array_schema, b);
+  CHECK(rc == TILEDB_OK);
+
+  // Check array schema
+  rc = tiledb_array_schema_check(ctx, array_schema);
+  CHECK(rc == TILEDB_OK);
+
+  // Create array
+  rc = tiledb_array_create(ctx, array_name.c_str(), array_schema);
+  CHECK(rc == TILEDB_OK);
+
+  // Clean up
+  tiledb_attribute_free(&a);
+  tiledb_attribute_free(&b);
+  tiledb_dimension_free(&d1);
+  tiledb_dimension_free(&d2);
+  tiledb_domain_free(&domain);
+  tiledb_array_schema_free(&array_schema);
+}
+
+TEST_CASE_METHOD(
+    QueryAggregateFx,
+    "C API: argument validation",
+    "[capi][query_aggregate][args]") {
```

Comment:
This is a very long test that does a lot of different argument validations... I think it can be split up.